source/server/hot_restart_impl.cc

```diff
@@ -109,22 +109,24 @@ void SharedMemory::initializeMutex(pthread_mutex_t& mutex) {
   pthread_mutex_init(&mutex, &attribute);
 }
 
-std::string SharedMemory::version(uint64_t max_num_stats, uint64_t max_stat_name_len) {
+std::string SharedMemory::version(uint64_t max_num_stats,
+                                  const Stats::StatsOptions& stats_options) {
   return fmt::format("{}.{}.{}.{}", VERSION, sizeof(SharedMemory), max_num_stats,
-                     max_stat_name_len);
+                     stats_options.maxNameLength());
```

Comment:
This change doesn't change the shared memory layout at all, right? Just being extra paranoid I would double check to make sure nothing changes w/ hot restart and with the output versions, as we have had issues in this area in the past. I'm not sure what our test status is on hot restart versions changing.

Reply:
It doesn't change that last argument to `fmt:;format()`, since that value `max_stat_name_len` used to be the sum of `max_obj_name_len` and `maxStatSuffixLength()`, and it still is. But I'll rerun the hot_restart_impl_test on this and upstream master and just make sure.

Reply:
I reran this test for master and this branch and the hot restart versionString is the same. :smile: In master right now, the `server/mocks.cc` sets `maxObjNameLength()` to return `150` by default. This new test does not, so the versionStrings look different -- but if we set

```stats_options_.max_obj_name_length_ = 150```

somewhere in `TEST_F(HotRestartImplTest, versionString)`, then the versionStrings are identical.

Reply:
This is somewhat unrelated to your change, but WDYT about actually committing the hot restart version string into a test somewhere and verifying that it is what we expect it to be? That would warn us about a change in the hot restart version that we didn't intend? Thoughts?

Reply:
(Perhaps this goes in the hot restart test itself)?

Reply:
I like this -- I could hardcode that string into `hotrestart_test.sh` for comparison, merge in that PR first, and then verify that this PR doesn't need any changes to pass that test.

Reply:
+1, thank you.

Reply:
@mattklein123 see #3832.